thunder/dynamo/compiler.py

```diff
@@ -36,27 +55,180 @@ def __init__(self, **thunder_options):
             ...         return x - 1
             >>> out = func(x)
         """
-        from thunder import ThunderModule
+        from thunder import ThunderModule, jit
 
         _warn_thunder_compiler()
 
         # Thunder-compiled functions should be readily available for inspection
-        # and testing, so we will store them in a list. The order of the
+        # and testing, so we will store them in a list[SubgraphInfo]. The order of the
         # functions in the list will be the same as the order in which they were
-        # compiled. In addition, we will store a mapping from the ThunderModule
-        # to the GraphModule that was passed to ThunderCompiler. This will allow
-        # us to inspect the GraphModule that was compiled by Thunder.
-        self.thunder_fns: list[ThunderModule] = []
-        self.thunder_to_gm: dict[ThunderModule, torch.fx.GraphModule] = {}
+        # compiled.
+        # Ref to the documentation of `SubgraphInfo` to know more about the information it contains.
+        self.subgraph_infos: list[SubgraphInfo] = []
+
+        if thunder_options is None:
+            thunder_options = {}
+
+        if torch_inductor_options is None:
+            torch_inductor_options = {}
 
         self.thunder_options = thunder_options
+        self._thunder_jit = partial(jit, **thunder_options)
+        self._torch_compile = partial(torch.compile, **torch_inductor_options)
+
+    def _splitter(
+        self, gm: torch.fx.GraphModule, _unused_sample_args: list[torch.SymInt, torch.Tensor]
+    ) -> torch.fx.GraphModule:
```

Comment:
This method doesn't seem to use the `self` attribute. Let's move to a separate file `splitter.py`.